nemo/lightning/resume.py

```diff
@@ -148,18 +140,52 @@ def nemo_path(self, model=None) -> Optional[Path]:
             else:
                 checkpoint = last_checkpoints[0]
 
+        if not checkpoint and self.restore_path:
+            return self._try_restore_model(model)
+
         if checkpoint:
             if self.adapter_path:
-                return AdapterPath(checkpoint, adapter_path=Path(self.adapter_path))
-            return Path(checkpoint)
+                return AdapterPath(Path(self.adapter_path), base_model_path=checkpoint), None
+            else:
+                # resuming a PEFT run, load
+                import json
+
+                from nemo.lightning.pytorch.callbacks.peft import _ADAPTER_META_FILENAME
+
+                adapter_meta_path = checkpoint / _ADAPTER_META_FILENAME
+                if adapter_meta_path.exists():
+                    with open(adapter_meta_path, "r") as f:
+                        metadata = json.load(f)
+                    if not self.restore_path:
+                        self.restore_path = metadata['model_ckpt_path']
+                    _, base_model_path = self._try_restore_model(model)
+
+                    if base_model_path != Path(metadata['model_ckpt_path']):
+                        raise ValueError(
+                            f"When trying to resume a PEFT training run, found mismatching values: "
+                            f"your specified restore_path points to {base_model_path}, "
+                            f"but the PEFT checkpoint was trained with "
+                            f"model_ckpt_path={metadata['model_ckpt_path']}"
+                        )
+
+                    # return AdapterPath(base_model_path, adapter_path=checkpoint), None
+                    return AdapterPath(checkpoint, base_model_path=base_model_path), None
```

Comment:
Can you move this to a separate private method?

Reply:
good idea